src/Modal/index.js

```diff
@@ -1,20 +1,36 @@
-import React from 'react'
+import React, { useEffect } from 'react'
 import {default as PT} from 'prop-types'
 import cx from 'classnames'
 import Icon from '../Icon'
 
-const Modal = ({ extraClass, children, onClose, ...props }) => {
+const Overlay = ({ children, open, ...props }) => open ?
+  <div className="ola_modal-overlay" {...props}>
+    {children}
+  </div> : null
+
+const Modal = ({ extraClass, children, closeWhenClickOut = true, closeWhenPressEsc = true, onClose, ...props }) => {
+    
+  useEffect(() => closeWhenPressEsc && document.body.addEventListener('keydown', e => e.keyCode === 27 && onClose()), [])
```

Comment:
- This useEffect should be shared for reused in other components. 
- If use addEventListener you should use removeEventListener to remove this listener.